test/ranges-test.cc

```diff
@@ -96,15 +100,17 @@ TEST(RangesTest, JoinInitializerList) {
 struct my_struct {
   int32_t i;
   std::string str;  // can throw
-  template <size_t N> decltype(auto) get() const noexcept {
-    if constexpr (N == 0)
-      return i;
-    else if constexpr (N == 1)
-      return fmt::string_view{str};
+  template <size_t N> fmt::enable_if_t<N == 0, int32_t> get() const noexcept {
+    return i;
+  }
+  template <size_t N>
+  fmt::enable_if_t<N == 1, fmt::string_view> get() const noexcept {
+    return {str};
```

Comment:
No C++14 `decltype(auto)` (probably was unneeded anyway) and C++17 `constexpr if`, only C++11 code